awscrt/s3.py

```diff
@@ -517,13 +545,18 @@ def _on_finish(self, error_code, status_code, error_headers, error_body):
         error = None
         if error_code:
             error = awscrt.exceptions.from_code(error_code)
-            if error_body:
-                # TODO The error body is XML, will need to parse it to something prettier.
-                try:
-                    extra_message = ". Body from error request is: " + str(error_body)
-                    error.message = error.message + extra_message
-                except BaseException:
-                    pass
+
+            # Make this into an S3ResponseError, if possible
+            if isinstance(error, awscrt.exceptions.AwsCrtError) \
+                    and status_code is not None \
+                    and error_headers is not None:
```

Comment:
Do we know if there won't be any case where status_code is `AWS_ERROR_S3_INVALID_RESPONSE_STATUS` but error_headers is None and error_body is present? Why assert on `error_headers` at all? I think we should just check if the status_code is `AWS_ERROR_S3_INVALID_RESPONSE_STATUS` and return whatever S3 gave us.

Reply:
I'll add better comment explaining my logic

Reply:
There are several aws-error-codes besides AWS_ERROR_S3_INVALID_RESPONSE_STATUS where the failure is due to a response